EosPayload/lib/orcheostrator/orcheostrator.py

```diff
@@ -84,64 +97,125 @@ def valid_driver(driver) -> bool:
         self._logger.info("Spawning Drivers")
-        for attribute_name in dir(drivers):
-            driver = getattr(drivers, attribute_name)
-            if self.valid_driver(driver):
-                container = DriverContainer(driver)
-                try:
-                    if driver.get_device_id() is None:
-                        self._logger.error(f"can't spawn process for device from class '{driver.__name__}'"
-                                           " because device id is not defined")
-                        container.update_status(Status.INVALID)
-                        self._drivers['<' + driver.__name__ + '>'] = container
-                        continue
-                    if not driver.enabled():
-                        self._logger.warning(f"skipping device '{driver.get_device_pretty_id()}' from"
-                                             f" class '{driver.__name__}' because it is not enabled")
-                        container.update_status(Status.DISABLED)
-                        self._drivers[driver.get_device_id()] = container
-                        continue
-                    self._logger.info(f"spawning process for device '{driver.get_device_pretty_id()}' from"
-                                      f" class '{driver.__name__}'")
-                    proc = Process(target=self._driver_runner, args=(driver, self.output_directory), daemon=True)
-                    container.process = proc
-                    container.update_status(Status.HEALTHY, 1)
-                    proc.start()
-                    self._drivers[driver.get_device_id()] = container
-                except Exception as e:
-                    self._logger.critical("A fatal exception occurred when attempting to load driver from"
-                                          f" class '{driver.__name__}': {e}\n{traceback.format_exc()}")
+        driver_list = self.config["enabled_drivers_list"]
+
+        for driver_tuple in driver_list:
+            driver = driver_tuple[0]
+            driver_config = driver_tuple[1]
+            container = DriverContainer(driver, config=driver_config)
+            try:
+                if driver_config.get("device_id") is None:
+                    self._logger.error(f"can't spawn process for device from class '{driver.__name__}'"
+                                       " because device id is not defined")
                     container.update_status(Status.INVALID)
                     self._drivers['<' + driver.__name__ + '>'] = container
+                    continue
+                self._logger.info(f"spawning process for device '{get_pretty_id(driver_config)}' from"
+                                  f" class '{driver.__name__}'")
+                proc = Process(target=self._driver_runner, args=(driver, self.output_directory, driver_config), daemon=True)
+                container.process = proc
+                container.update_status(Status.HEALTHY, 1)
```

Comment:
there should probably be an INITIALIZED or SPAWNED state for when drivers are first launched

Reply:
Done. There is a slight change of behavior in that devices are only reported HEALTHY after orchEOStrator has actually gotten a health report, which I think is an improvement.